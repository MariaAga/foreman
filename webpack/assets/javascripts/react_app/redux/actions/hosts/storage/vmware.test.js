import {
  datastoresUrl,
  storagePodsUrl,
  basicConfig,
  initAction,
  changeClusterAction,
  state1,
  fetchDatastoreParams,
  fetchStoragePodsParams,
} from './vmware.fixtures';

import * as actions from './vmware';

describe('vmware storage hosts actions', () => {
  describe('initController', () => {
    it('initializes the container', () => {
      const dispatch = jest.fn();
      const dispatcher = actions.initController(
        basicConfig,
        'cluster',
        null,
        null
      );

      dispatcher(dispatch);

      expect(dispatch).toHaveBeenCalledTimes(3);
      expect(dispatch).toHaveBeenCalledWith(initAction);
    });
  });

  describe('changeCluster', () => {
    it('changes the cluster and refetches the storages', () => {
      const dispatch = jest.fn();
      const dispatcher = actions.changeCluster('newCluster');

      dispatcher(dispatch, () => state1);

      expect(dispatch).toHaveBeenCalledTimes(3);
      expect(dispatch).toHaveBeenCalledWith(changeClusterAction);
    });
  });

  describe.each([
    ['fetchDatastores', datastoresUrl, fetchDatastoreParams],
    ['fetchStoragePods', storagePodsUrl, fetchStoragePodsParams],
  ])('%s', (actionName, url, fetchParams) => {
    it('doesnt make the ajax request when cluster is not set', () => {
      const dispatcher = actions[actionName](url, null);
      expect(dispatcher).toMatchSnapshot();
    });

    it('makes the ajax request to the right url', () => {
      const dispatcher = actions[actionName](url, 'cluster');
      expect(dispatcher).toMatchSnapshot();
    });
  });
});
