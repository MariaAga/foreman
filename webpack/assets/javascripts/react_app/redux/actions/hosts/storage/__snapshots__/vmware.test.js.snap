// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vmware storage hosts actions fetchDatastores doesnt make the ajax request when cluster is not set 1`] = `Object {}`;

exports[`vmware storage hosts actions fetchDatastores makes the ajax request to the right url 1`] = `
Object {
  "payload": Object {
    "params": Object {
      "cluster_id": "cluster",
    },
  },
  "subtype": "STORAGE_VMWARE_DATASTORES",
  "type": "API_GET",
  "url": "test.com/datastores",
}
`;

exports[`vmware storage hosts actions fetchStoragePods doesnt make the ajax request when cluster is not set 1`] = `Object {}`;

exports[`vmware storage hosts actions fetchStoragePods makes the ajax request to the right url 1`] = `
Object {
  "payload": Object {
    "params": Object {
      "cluster_id": "cluster",
    },
  },
  "subtype": "STORAGE_VMWARE_STORAGEPODS",
  "type": "API_GET",
  "url": "test.com/storage_pods",
}
`;
